lib/ruby-push-notifications/mpns/mpns_result.rb

```diff
@@ -0,0 +1,182 @@
+
+module RubyPushNotifications
+  module MPNS
+    # Class that encapsulates the result of a single sent notification to a single
+    # Device URL
+    # (https://msdn.microsoft.com/en-us/library/windows/apps/ff941100%28v=vs.105%29.aspx)
+    class MPNSResult
+      # @return [String]. Receiver MPNS device URL.
+      attr_accessor :device_url
+
+      # @private X-NotificationStatus HTTP Header string
+      X_NOTIFICATION_STATUS  = 'x-notificationstatus'
```

Comment:
It is because each_header method convert key values to downcase.. The only way to read response header in case-sensitive mode is use the response http .. but it is hard to test because is not possible initialize a Net::Response with a fake hash unless create a post request and use webmock to return headers but it makes the tests become very confusing.
http://ruby-doc.org/stdlib-2.2.1/libdoc/net/http/rdoc/Net/HTTPHeader.html#method-i-each_header